openapi3gen/openapi3gen.go

```diff
@@ -104,8 +129,13 @@ func (g *Generator) NewSchemaRefForValue(value interface{}, schemas openapi3.Sch
 		return nil, err
 	}
 	for ref := range g.SchemaRefs {
-		if _, ok := g.componentSchemaRefs[ref.Ref]; ok && schemas != nil {
-			schemas[ref.Ref] = &openapi3.SchemaRef{
+		refName := ref.Ref
+		if g.opts.exportComponentSchemas.ExportComponentSchemas && strings.HasPrefix(ref.Ref, "#/components/schemas/") {
+			refName = strings.TrimPrefix(ref.Ref, "#/components/schemas/")
+		}
+
+		if _, ok := g.componentSchemaRefs[refName]; ok && schemas != nil {
+			schemas[refName] = &openapi3.SchemaRef{
```

Comment:
There is a problem here with the schema customizer... if we have the same schema called MyObject that we have already parsed for one path and then another path has that same object but it has a schema customizer that adds a field it overrides the existing object for the previous path... Is the solution here just that the user needs to have two different structs? As far as I know there is no way to extend a schema within an operation?

Reply:
@EnriqueL8 I'm not sure what you meant.

May I ask you please to provide a couple of samples so that I could help you?

Reply:
So if you have two different paths in your api server:
`POST /object -> returns an object`
`PATCH /object -> returns an object with an extra field that contains an array of changes made by the patch`
If the object in Golang is declared as:
```
type Object struct {
     field1 string `json:"field1"`
     field2 string `json:"field2"`
     patchedFields []string `json:"patchedFields,omitempty"`
}
```

Then when the OpenAPI document gets generated the second `PATCH` operation will override `#/components/schemas/object` and add the `patchedFields` field and it will appear in the `POST /object` operation.

It would be the same case if a user supplies an schema customizer and adds a field at that point.

The solution could be that it's opinionated and they should be two structs:
```
type Object struct {
     field1 string `json:"field1"`
     field2 string `json:"field2"`
}
```
```
type ObjectWithPatchedFields struct {
     Object
     patchedFields []string `json:"patchedFields,omitempty"`
}
```


I don't if there is another way?

Reply:
@EnriqueL8 , I think it is not a technical problem. A developer must be aware and don't create different structs with the same name. Also, it is already possible to customize to component names with the names generator. I think it is already flexible enough.

What do you think on that?

Reply:
I'm wondering if a better solution would be to merge schemas on the schema customiser if it exists as a component?

Reply:
I'm thinking specifically for things such as `readOnly` and `writeOnly` or any other properties on the schema that need to be preserved

Reply:
Maybe https://github.com/getkin/kin-openapi/pull/920 can help here ;)